src/main/java/ArraySortedList.java

```diff
@@ -0,0 +1,184 @@
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class ArraySortedList<T> implements SortedList<T> {
```

Comment:
Thinking more about this, a `SortedList` isn't a [List](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html) in the java sense.

> An ordered collection (also known as a sequence). **The user of this interface has precise control over where in the list each element is inserted.** The user can access elements by their integer index (position in the list), and search for elements in the list.

Emphasis mine.

These are, however, [bags](https://en.wikipedia.org/wiki/Multiset). Think at the very least, a rename is in order. Though, changing the `List` interface up would make more sense, since `first` and `last` don't really make sense for a bag. You could ask for multiplicity, total size, and add/remove multiple copies of objects. Eg, [guava's Multiset](https://guava.dev/releases/30.1.1-jre/api/docs/com/google/common/collect/Multiset.html).

Reply:
Brilliant!